### PR TITLE
Refs #33697 -- Made MultiPartParser use django.utils.http.parse_header_parameters() for parsing Content-Type header.

### PR DESCRIPTION
**Pull request by mehrdadmoradii**
Switching to parse_header_parameters() in the MultiPartParser.__init__(), in order to refactor in the following step.

### CODE REVIEW

File: django/http/multipartparser.py [hunk 1]
```diff
@@ -67,17 +68,21 @@ def __init__(self, META, input_data, upload_handlers, encoding=None):
...
-            ctypes, opts = parse_header(content_type.encode("ascii"))
+            content_type.encode("ascii")
         except UnicodeEncodeError:
             raise MultiPartParserError(
                 "Invalid non-ASCII Content-Type in multipart: %s"
                 % force_str(content_type)
             )
+
+        if not content_type.startswith("multipart/"):
+            raise MultiPartParserError("Invalid Content-Type: %s" % content_type)
```
Comment (felixxm) [inline django/http/multipartparser.py:81]: Is there a reason to move this check? I'd move it to the previous place.

Comment (mehrdadmoradii) [reply django/http/multipartparser.py:81]: Thanks for mentioning that. I'll move it to it's previous place.

**Comment by felixxm** [comment]
@mehrdadmoradii Thanks :+1:

**Comment by mehrdadmoradii** [comment]
@felixxm Thanks Mariusz. What would be the next step?

**Pull request merged**